control-plane/agents/src/bin/core/main.rs

```diff
@@ -150,6 +150,12 @@ pub(crate) struct CliArgs {
     /// Allow pools to be created using non persistent devlinks.
     #[clap(long)]
     allow_non_persistent_devlink: bool,
+
+    /// Prefer to use encrypted pools for volume replicas if the this is set in addition to
+    /// encryption on volume spec. Applicable if the volume was initially provisioned via
+    /// a non-encryption storage class.
+    #[clap(long)]
+    prefer_encrypted_pools: bool,
```

Comment:
nit: parameter name itself could be misunderstood, maybe something more like this:
`encrypted_pools_soft_scheduling` ?